smart_contract/src/karma_tiers_test.rs

```
@@ -0,0 +1,348 @@
+                    println!(
+                        "Tier {}: {} ({} - {}) - {} tx/epoch",
+                        i, tier.name, tier.minKarma, tier.maxKarma, tier.txPerEpoch
+                    );
+                }
+                Err(e) => {
+                    eprintln!("Failed to get tier {}: {}", i, e);
+                }
+            }
+        }
+    }
+
+    // Test 3: Create tiers
+    let test_tiers = vec![
+        KarmaTiersSC::Tier {
+            minKarma: U256::from(0),
+            maxKarma: U256::from(10),
+            name: "NoTier".to_string(),
+            txPerEpoch: 6,
+        },
+        KarmaTiersSC::Tier {
+            minKarma: U256::from(10),
+            maxKarma: U256::from(49),
+            name: "Basic".to_string(),
+            txPerEpoch: 6,
+        },
+        KarmaTiersSC::Tier {
+            minKarma: U256::from(50),
+            maxKarma: U256::from(99),
+            name: "Active".to_string(),
+            txPerEpoch: 120,
+        },
+        KarmaTiersSC::Tier {
+            minKarma: U256::from(100),
+            maxKarma: U256::from(499),
+            name: "Regular".to_string(),
+            txPerEpoch: 720,
+        },
+        KarmaTiersSC::Tier {
+            minKarma: U256::from(500),
+            maxKarma: U256::from(999),
+            name: "Power User".to_string(),
+            txPerEpoch: 86400,
+        },
+        KarmaTiersSC::Tier {
+            minKarma: U256::from(1000),
+            maxKarma: U256::from(4999),
+            name: "S-Tier".to_string(),
+            txPerEpoch: 432000,
+        },
+    ];
+
+    println!(
+        "Attempting to update tiers with {} new tiers:",
+        test_tiers.len()
+    );
+    for (i, tier) in test_tiers.iter().enumerate() {
+        println!(
+            "  Tier {}: {} ({} - {}) - {} tx/epoch",
+            i, tier.name, tier.minKarma, tier.maxKarma, tier.txPerEpoch
+        );
+    }
+
+    if !PRIVATE_KEY.is_empty() && curren_tier_count_u8 == 0 {
```

This will fail silently if private_key is empty no?